### PR TITLE
Update haproxy, support gzip, fix <1024 ports

### LINKED ISSUE #1704: Support gzip for haproxy => save bandwith + time
```
make USE_ZLIB=yes
```

If haproxy is compiled using USE_ZLIB=yes it is possible to use gzip to compress a lot of data.
For example you could add this to your default config section:

```
 compression algo gzip
 compression type text/html text/plain text/css
```

There are many more compression types available.

### LINKED ISSUE #1730: Haproxy config doesn't support binding to privileged ports
A couple small tweaks need to be made to support this:

Raise privileges at runtime to bind to low ports ->
/var/packages/haproxy/scripts/start-stop-status:
```
# Others
INSTALL_DIR="/usr/local/${PACKAGE}"
PYTHON_DIR="/usr/local/python"
PATH="${INSTALL_DIR}/bin:${INSTALL_DIR}/env/bin:${PYTHON_DIR}/bin:/usr/local/bin:/bin:/usr/bin:/usr/syno/bin"
RUNAS="root"   <----Modification
PYTHON="${INSTALL_DIR}/env/bin/python"
HAPROXY="${INSTALL_DIR}/sbin/haproxy"
PID_FILE="${INSTALL_DIR}/var/haproxy.pid"
CFG_FILE="${INSTALL_DIR}/var/haproxy.cfg"
```

Drop privileges for security->
haproxy.cfg.tpl:
```
global
        daemon
        maxconn 256
        user haproxy <----Modification
        log localhost user info
        spread-checks 10
        tune.ssl.default-dh-param 2048
```

<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/19006515-haproxy-config-doesn-t-support-binding-to-privileged-ports?utm_campaign=plugin&utm_content=tracker%2F282943&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F282943&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### LINKED ISSUE #1782: [HAPROXY] 1.5.14 Security Update...
Can you update the SynoCommunity HAProxy Package for security reasons ? 

>July, 3rd, 2015 : 1.5.14 : fixes an information leak vulnerability (CVE-2015-3281)

>A vulnerability was found when HTTP pipelining is used. In some cases, a client might be able to cause a buffer alignment issue and retrieve uninitialized memory contents that exhibit data from a past request or session. I want to address sincere congratulations to Charlie Smurthwaite of aTech Media for the really detailed traces he provided which made it possible to find the cause of this bug. Every user of 1.5-dev, 1.5.x or 1.6-dev must upgrade to 1.5.14 or latest 1.6-dev snapshot to fix this issue, or use the backport of the fix provided by their operating system vendors. CVE-2015-3281 was assigned to this bug. Code and changelog are available here as usual.

source : http://www.haproxy.org/

Thanks for your work...

<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/23748543-haproxy-1-5-14-security-update?utm_campaign=plugin&utm_content=tracker%2F282943&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F282943&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by piejanssens**
Fixes #1704
Fixes #1730 
Fixes #1782
Updates haproxy to latest stable version (from 1.5.11 to 1.5.14)

### CODE REVIEW

**Comment by Diaoul** [comment]
LGTM

**Comment by Dr-Bean** [comment]
Does the haproxy.cfg change work for existing installations? Seems like it might be overwritten during postupgrade, no?

**Comment by piejanssens** [comment]
Good point. Where can we add the new line for haproxy.cfg for existing installations?

**Comment by piejanssens** [comment]
I guess I could use "postupgrade ()" in installer.sh.
Would this work?https://github.com/piejanssens/spksrc/commit/46ed6cd33f32caa902f27cd887d1a0959863ba82

**Comment by Diaoul** [comment]
True, that would cause some trouble. Need to add a migration script.

**Comment by piejanssens** [comment]
@Diaoul Should I undo my last commit?

**Comment by Diaoul** [comment]
Yes please remove this and make a migration script that will add the appropriate config section like migrations here: https://github.com/SynoCommunity/spksrc/blob/master/spk/haproxy/src/installer.sh#L88-L96

**Comment by piejanssens** [comment]
@Diaoul @Dr-Bean Done :+1:

**Comment by piejanssens** [comment]
It does add the correct line in the correct spot, however, I couldn't get a TAB printed out by sed, so it hasn't got the correct indentation. I tried yesterday for over an hour using various methods.. 
Do you guys know of an spk where sed was already used to print a TAB?

**Comment by MattPark** [comment]
print the tab with \t
if it doesn't work then you need to add -r like `sed -i -r 's/tab/\t/'

**Comment by piejanssens** [comment]
The \t is not recognised as a tab in the 'replace by' part of sed.
Can you try it yourself with a test file on your syno ?

File: spk/haproxy/src/haproxy.cfg [hunk 1]
```diff
@@ -1,4 +1,5 @@
 global
+    user haproxy
```
Comment (Dr-Bean) [inline spk/haproxy/src/haproxy.cfg]: Seems like it should be a 4-space tab here, Github converts it to 8.

**Comment by Dr-Bean** [comment]
This might work: `awk 'NR == 2 {print "\tuser haproxy"} {print}' haproxy.cfg && mv haproxy.cfg.new haproxy.cfg`
Awk has it's downsides (...like no in-place editing, hence the `mv`), but it can pass a `\t` ;)

**Comment by piejanssens** [comment]
Got it! phew :smile:

**Comment by Dr-Bean** [comment]
That's not a tab?

**Comment by piejanssens** [comment]
@Dr-Bean It sure is. Used the tab key on my keyboard after the backslash. 
Tested the bash script on my syno and the result was a correctly tabbed entry in the haproxy.cfg file.

**Comment by Dr-Bean** [comment]
Yep, that works. Two things remaining: Make sure you entered a tab in the `haproxy.cfg` file (see my open comment on the code), because your editor put spaces in there as well. This handles new installations of the package.
You'll also need to update `haproxy.cfg.tpl` in the migration script, otherwise the change will be reverted as soon as the config is updated via the haproxy UI.

**Comment by piejanssens** [comment]
Indeed, forgot about the default haproxy.cfg. I believe this file is actually copied to haproxy.cfg.tpl, so there is no haproxy.cfg.tpl file itself.

**Comment by piejanssens** [comment]
@Dr-Bean I think we're good now :smile:

**Comment by piejanssens** [comment]
Any idea when this would be merged and deployed?

**Comment by Dr-Bean** [comment]
LGTM, so it'll be merged when possible. We don't give out ETA's...spare time and all.

**Comment by piejanssens** [comment]
@Dr-Bean @Diaoul Guys, can you check something out for me please?
I compiled this package to install on my Syno (DS413), but when I install it using the package manager (manual install) I get the message "Invalid file format".. I have no clue what could be wrong. The architecture is correct and there was no error whilst building the spk...

Here is my package: https://dl.dropboxusercontent.com/u/3691492/haproxy_qoriq-5.1_1.5.14-18.spk

**Comment by MattPark** [comment]
I've been getting that for a lot of packages lately.  It still seems to
install just fine though

**Comment by piejanssens** [comment]
In my case it didn't install unfortunately.
Btw the current haproxy pkg has version 1.5.12-18. 
Is the 18 the spk revision? Because I bumped from 17 to 18 in my commits. 
So I'm confused why the repo would contain spk revision 18 already...

**Comment by Dr-Bean** [comment]
I think @Diaoul updated the package to fix something, but the update wasn't pushed to the repo. @piejanssens, you can set the version to 19...or we do that after merging this PR, either way works.

As for the package installation failures, that's a browser issue then? I'd suggest to open a ticket with Synology for that.

**Comment by piejanssens** [comment]
@Dr-Bean Bumped to 19 - good to go. Don't bother with the installation issues I had, it was a problem caused by my VM (Parallels) while copying the archive over to my host system (OS X).

**Comment by Diaoul** [comment]
Compiling now, this should take quite some time.

**Comment by Diaoul** [comment]
Published.

**Pull request merged**